leakcanary-analyzer/src/test/java/com/squareup/leakcanary/HahaHelperTest.java

```diff
@@ -0,0 +1,239 @@
+package com.squareup.leakcanary;
+
+import com.squareup.haha.perflib.ArrayInstance;
+import com.squareup.haha.perflib.ClassInstance;
+import com.squareup.haha.perflib.ClassObj;
+import com.squareup.haha.perflib.Field;
+import com.squareup.haha.perflib.Snapshot;
+import com.squareup.haha.perflib.Type;
+import com.squareup.haha.perflib.io.HprofBuffer;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class HahaHelperTest {
+  public static final int STRING_CLASS_ID = 100;
+  public static final int CHAR_ARRAY_CLASS_ID = 101;
+  public static final int STRING_INSTANCE_ID = 102;
+  public static final int VALUE_ARRAY_INSTANCE_ID = 103;
+
+  public static final int VALUE_ARRAY_LENGTH = 6;
+  public static final int COUNT_VALUE = 5;
+  public static final int OFFSET_VALUE = 1;
+
+  private FakeHprofBuffer buffer;
+  private Snapshot snapshot;
+  private ClassInstance stringInstance;
+
+  @Before
+  public void setUp() {
+    buffer = new FakeHprofBuffer();
+    buffer.setStringsToRead("abcdef");
+
+    snapshot = new Snapshot(buffer);
+    snapshot.setIdSize(4);
```

Comment:
What does this do? Is that in other tests too? Seems weird, should be shared in test helper code and commented.